jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/jbpm/services/CustomKModuleDeploymentService.java

```diff
@@ -0,0 +1,31 @@
+package org.jbpm.test.performance.jbpm.services;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jbpm.kie.services.impl.KModuleDeploymentService;
+import org.jbpm.kie.services.impl.KModuleDeploymentUnit;
+import org.jbpm.services.api.model.DeployedUnit;
+import org.kie.api.runtime.KieContainer;
+import org.kie.api.runtime.manager.RuntimeEnvironmentBuilder;
+import org.kie.internal.runtime.conf.MergeMode;
+
+public class CustomKModuleDeploymentService extends KModuleDeploymentService {
+
+    protected Map<String, Object> customEnvironmentEntries = new HashMap<>();
+
+    public void addEnvironmentEntry(String name, Object value) {
+        customEnvironmentEntries.put(name, value);
+    }
+
+    @Override
+    protected RuntimeEnvironmentBuilder boostrapRuntimeEnvironmentBuilder(KModuleDeploymentUnit deploymentUnit,
+                                                                          DeployedUnit deployedUnit, KieContainer kieContainer, MergeMode mode) {
+        RuntimeEnvironmentBuilder builder = super.boostrapRuntimeEnvironmentBuilder(deploymentUnit, deployedUnit, kieContainer, mode);
+        for (Map.Entry<String, Object> envEntry : customEnvironmentEntries.entrySet()) {
+            builder.addEnvironmentEntry(envEntry.getKey(), envEntry.getValue());
+        }
+        return builder;
```

Comment:
BTW for this you could just override the method [AbstractKieServicesTest#createDeploymentDescriptor()](https://github.com/kiegroup/jbpm/blob/master/jbpm-test/src/main/java/org/jbpm/test/services/AbstractKieServicesTest.java#L139) and also createDescriptor() to return true. But up to you.

Reply:
I think I'm gonna leave it as is for the moment cause I'm running out of time, but yes, it's better than my approach. Thanks for that